router/event.go

```diff
@@ -63,10 +65,20 @@ func (router *Router) eventFromRequest(r *http.Request) (*eventpkg.Event, string
 	}
 
 	event := eventpkg.New(eventType, mime, body)
-	if mime == mimeJSON && len(body) > 0 {
-		err = json.Unmarshal(body, &event.Data)
-		if err != nil {
-			return nil, "", errors.New("malformed JSON body")
+
+	// Because event.Data is []bytes here, it will be base64 encoded by default when being sent to remote function,
+	// which is why we change the event.Data type to "string" for forms, so that, it is left intact.
+	if len(body) > 0 {
+		switch {
+		case mime == mimeJSON:
+			err := json.Unmarshal(body, &event.Data)
+			if err != nil {
+				return nil, "", errors.New("malformed JSON body")
+			}
+			break
```

Comment:
@elsteelbrain `break` isn't needed in Golang case statements -- https://tour.golang.org/flowcontrol/9